src/mono/wasm/runtime/jiterpreter-trace-generator.ts

```diff
@@ -138,7 +138,32 @@ function is_backward_branch_target (
     return false;
 }
 
-export function generate_wasm_body (
+export function scanForLdlocas (
+    startOfBody: MintOpcodePtr, ip: MintOpcodePtr, endOfBody: MintOpcodePtr
+) {
+    const end = Math.min(<any>ip, <any>endOfBody);
+    ip = startOfBody;
+
+    while (<any>ip < end) {
+        const opcode = getU16(ip);
+        const info = OpcodeInfo[opcode];
+
+        switch (opcode) {
+            case MintOpcode.MINT_LDLOCA_S:
+                addressTakenLocals.add(getArgU16(ip, 2));
```

Comment:
Why is `addressTakenLocals` also populated in `append_ldloca`

Reply:
This first pass runs for the portion of the method preceding a trace. i.e. we start compiling at offset 200 in the method, this pass runs from 0-200 to pre-populate the table, then the normal trace generator runs for the rest

Reply:
I'm quite confused now. You are populating the table of address taken locals from opcodes other than `MINT_LDLOCA`. How exactly does all this work ?

Also the fact that you don't fully scan for `MINT_LDLOCA` at the beginning doesn't sound right. I assume you use this table on the fly for some optimizations, and you populate it again with locals, but maybe the local has its address taken towards the end of the method code and then jumps back to the beginning of the method where the jiterpreter code doesn't know about this.

Reply:
It sounds like I need to scan the entire method. That makes sense, and I'm not doing it yet. I'll make the switch.

While all calls to append_ldloca can theoretically populate it, in practice LDLOCA is the only one that does (I checked) - this is just there so that any code using it without manual annotations (since the code used to not contain this info at all) will by default mark it as leaking the address. So as far as that is concerned, the scan only caring about LDLOCA is correct. append_ldloca has a slightly messy signature where you either tell it 'i'm invalidating X bytes and the address is/isn't visible after this opcode (i.e. ldloca)' or 'i'm invalidating 0 bytes and the address doesn't leak'